### ISSUE TITLE
Warning: Closing a shared STX with nonzero ref. count

### ISSUE THREAD

**Issue by davidozog** [Sandia-OpenSHMEM/SOS#662]
I've seen this warning on PSM2 (albeit very rarely, and only with threaded tests).  
```
[0001] WARN:  ../../src/transport_ofi.c:1679: shmem_transport_fini
[0001]        Closing a shared STX (1) with nonzero ref. count (1)
```
So far, the best way to reproduce is with the nonblocking multi-threaded bandwith test:
```
SHMEM_OFI_STX_DISABLE_PRIVATE=1 SHMEM_OFI_STX_MAX=5 oshrun --bind-to core:8 -hosts node3,node4 -ppn 1 ~/SOS/test/performance/shmem_perf_suite/shmem_bw_put_ctx_perf_nb -T 8 -C MULTIPLE
```
After the final 8GB run I see the warning fairly consistently (~80%).  On my platform (Neptune), the warning does not seem to occur with STX_MAX=4 or less, but does seem to occur with STX_MAX>5.

**Comment by jdinan** [comment]
Looks like we are missing mutex lock/unlock around accesses to the STX structures in `shmem_transport_ctx_destroy`.

**Comment by davidozog** [comment]
Yup, that looks to be the problem.

**Comment by davidozog** [comment]
Closed by #663
